sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -279,11 +279,13 @@ class CodegenContext {
       inlineToOuterClass: Boolean = false): String = {
     // The number of named constants that can exist in the class is limited by the Constant Pool
     // limit, 65,536. We cannot know how many constants will be inserted for a class, so we use a
-    // threshold of 1600k bytes to determine when a function should be inlined to a private, nested
-    // sub-class.
+    // threshold to determine when a function should be inlined to a private, nested sub-class
+    val generatedClassLengthThreshold = SparkEnv.get.conf.getInt(
```

Comment:
Can you ensure `SparkEnv.get` is always true?

Reply:
Can this be empty? In https://github.com/apache/spark/blob/83488cc3180ca18f829516f550766efb3095881e/sql/catalyst/src/main/java/org/apache/spark/sql/execution/UnsafeExternalRowSorter.java#L80 there is no check about it being `None`...

Reply:
The code that you pointed out is not in `CodeGenerator.scala`. In [my case](https://github.com/apache/spark/pull/18966/files#diff-8bcc5aea39c73d4bf38aef6f6951d42cR776) at `CodeGenerator.scala`, there are some cases that `SparkEnv.get` is `null`. If you have not seen this `null` in you case, it would be good.
@gatorsmile will implement a better approach to get this conf soon.

Reply:
you better refer other code in `CodeGenerate`:

https://github.com/mgaido91/spark/blob/c69be31314d9aa96c3920073beaf7cca46d507fa/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala#L934